src/legacy/ui/public/agg_types/controls/time_interval.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get, find } from 'lodash';
+import React, { useEffect } from 'react';
+
+import { EuiFormRow, EuiIconTip, EuiComboBox, EuiComboBoxOptionProps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { AggParamEditorProps } from '../../vis/editors/default';
+import { AggParamOption } from '../agg_param';
+import { parseInterval } from '../utils';
+
+interface ComboBoxOption extends EuiComboBoxOptionProps {
+  key: string;
+}
+
+function TimeIntervalParamEditor({
+  agg,
+  aggParam,
+  editorConfig,
+  value,
+  setValue,
+  showValidation,
+  setValidity,
+}: AggParamEditorProps<string>) {
+  const timeBase: string = get(editorConfig, 'interval.timeBase');
+  const options = timeBase
+    ? []
+    : (aggParam.options || []).reduce(
+        (filtered: ComboBoxOption[], option: AggParamOption) => {
+          if (option.enabled ? option.enabled(agg) : true) {
+            filtered.push({ label: option.display, key: option.val });
+          }
+          return filtered;
+        },
+        [] as ComboBoxOption[]
+      );
+
+  let selectedOptions: ComboBoxOption[] = [];
+  let definedOption: ComboBoxOption | undefined;
+  if (value) {
+    definedOption = find(options, { key: value });
+    selectedOptions = definedOption ? [definedOption] : [{ label: value, key: 'custom' }];
+  }
+
+  const isValid = value ? (definedOption ? true : parseInterval(value, timeBase)) : false;
```

Comment:
This expression is confusing to me. Maybe you could rewrite as:

`const isValid = value && !!(definedOption || parseInterval(value, timeBase);`

or alternately, I see you already have `if (value)` on line 58- maybe you could use that?

Reply:
Updated.